compiler/src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -1012,5 +1011,21 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
     val encoding = ctx.settings.encoding.value
     if (file != null && file.exists) new SourceFile(file, Codec(encoding)) else NoSource
   }
+
+  /** Desugar identifier into a select node. Return None if not possible */
```

Comment:
We don't usually use Option in the compiler, first, because of the allocation overhead, and second, because we usually have sentinel values that signal absence of a proper result. E.g. `EmptyTree` for trees. So I would just let `desugarIdent` return an `EmptyTree` if no rules apply.  That's more idiomatic. Hint: `orElse` works on empty trees.

I see there are quite a lot of uses of `desugarIdent`, and have the impression that most of these uses could be improved by going to EmptyTree. The use of Option.fold in JSCodeGen is particularly opaque.